File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        # keep an index for travrsign the array
        # keep an ele for markign the most repeated value
        count = 1
        cand = nums[0]

        if len(nums)<=1:
            return nums[0]

        for x in range(1,len(nums)):
            if nums[x] != cand:
                if count<1:
                    cand = nums[x]
                else:
                    count-=1
            else:
                count+=1
        return cand

                


        

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cand = nums[0];
        int count = 1;

        for(int i=1; i < nums.length; i++){
            if(nums[i] != cand){
                if(count == 0){
                    cand = nums[i];
                }
                else{
                    count--;
                }
            }
            else{
                count++;
            }
        }

        return cand;
        
    }
}